tests/system_test.py

```diff
@@ -1107,15 +1107,10 @@ def on_accepted(self, event):
         event.delivery.settle()
 
     def on_released(self, event):
-        # for some reason Proton 'helpfully' calls on_released even though the
-        # delivery state is actually MODIFIED
         if event.delivery.remote_state == Delivery.MODIFIED:
-            return self.on_modified(event)
```

Comment:
This makes the handlers work the way they "should've been" done in Proton, imo. I thought that it was a reasonable improved thing to do. But ok, the Proton orthodoxy prevails.

Reply:
You know what, I agree with you. Leave it as is.